README.md

```diff
@@ -4,6 +4,262 @@ A trivial implementation of timeouts for `Promise`s, built on top of [React PHP]
 
 > Note: This project is in early alpha stage! Feel free to report any issues you encounter.
 
+## Usage
+
+This lightweight library consists only of a few simple functions.
+All functions reside under the `Clue\Promise\Timer` namespace.
+
+The below examples assume you use an import statement similar to this:
+
+```php
+use Clue\Promise\Timer;
+
+Timer\timeout(…);
+```
+
+Alternatively, you can also refer to them with their fully-qualified name:
+
+```php
+\Clue\Promise\Timer\timeout(…);
+``` 
+
+### timeout()
+
+The `timeout(PromiseInterface $promise, $time, LoopInterface $loop)` function
+can be used to *cancel* operations that take *too long*.
+You need to pass in an input `$promise` that represents a pending operation and timeout parameters.
+It returns a new `Promise` with the following resolution behavior:
+
+* If the input `$promise` resolves before `$time` seconds, resolve the resulting promise with its fulfillment value.
+* If the input `$promise` rejects before `$time` seconds, reject the resulting promise with its rejection value.
+* If the input `$promise` does not settle before `$time` seconds, *cancel* the operation and reject the resulting promise with a [`TimeoutException`](#timeoutexception).
+
+A common use case for handling only resolved values looks like this:
+
+```php
+$promise = accessSomeRemoteResource();
+Timer\timeout($promise, 10.0, $loop)->then(function ($value) {
+    // the operation finished within 10.0 seconds
+});
+```
+
+A more complete example could look like this:
+
+```php
+$promise = accessSomeRemoteResource();
+Timer\timeout($promise, 10.0, $loop)->then(
+    function ($value) {
+        // the operation finished within 10.0 seconds
+    },
+    function ($error) {
+        if ($error instanceof Timer\TimeoutException) {
+            // the operation has failed due to a timeout
+        } else {
+            // the input operation has failed due to some other error
+        }
+        throw $error;
+    }
+);
+```
+
```

Comment:
For React/Promise >= 2.2 this can be:

```php
Timer\timeout($promise, 10.0, $loop)
	->then(function ($value) {
        // the operation finished within 10.0 seconds
    })
    ->otherwise(function (Timer\TimeoutException $error) {
        // the operation has failed due to a timeout
    })
    ->otherwise(function ($error) {
        // the input operation has failed due to some other error
    })
;
```

Note that i intentionally omitted rethrowing the exception.